Migrate: progress-dashboard

lachlan-robinson opened the pull request:
# Description

This migration is based from 9.x

Migration of progress dashboard to angular17, typescript, tailwind & material.

## Type of change
- [x] Migration -> No change to functionality

# How Has This Been Tested?

Manual Testing:

- [x] Task Status displays the correct number of tasks and their status relative to the target grade.
- [x] The burndown chart updates to the target grade.
- [x] Progress dashboard is hidden when a task is selected and vice versa.
- [x] Tutor view header updates for the student dashboard being viewed.

Note:
- There appears to be a bug with the progress burndown chart that was pre-existing. The chart does not correctly display submitted and completed tasks.

## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari
- [x] Tested in latest Firefox

## Before

![progress-dashboard](https://github.com/user-attachments/assets/8d9d6232-32a4-41a0-bb15-52d57d609a45)

## After

![progress-dashboard-migrated](https://github.com/user-attachments/assets/ba2001b2-391c-458e-8e40-9d2d293a94d1)

## Tutor View

![tutor-view](https://github.com/user-attachments/assets/1b1a483c-a374-4586-ac17-61abd4675d17)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

lachlan-robinson committed:
chore: 9.x windows

lachlan-robinson committed:
chore: create new component files

lachlan-robinson committed:
chore: unlink old component + link new component

lachlan-robinson committed:
chore: migrate functionality + update to tailwindcss and material UI

lachlan-robinson committed:
chore: remove old component files

lachlan-robinson committed:
chore: responsive columns for visualisations

lachlan-robinson committed:
Update package-lock.json

lachlan-robinson committed:
Update package.json

lachlan-robinson committed:
Update package.json

lachlan-robinson committed:
Update package-lock.json

lachlan-robinson committed:
chore: update styling to better match original

lachlan-robinson committed:
chore: add custom breakpoint for responsive change

lachlan-robinson committed:
chore: tweak breakpoint for better visualisation visibility

lachlan-robinson committed:
chore: update tutor view functionality + tweak event emitter name

LovleenKala commented:
Hi Lachlan!

I was able to successfully pull your PR #305 into my local environment.
I tested it on my MacOS system without any build or installation issues.
After starting the application, the dashboard loaded correctly, including:
I tested the unit selection via the dropdown and verified that the assignments list, target grade selector, and progress graph are all working correctly.
The progress graph rendering properly with the expected data points.
The unit selection and student information displaying as intended.
I did not encounter any console errors or broken UI behavior during my testing.
The functionality matched the description provided in the PR and integrated well into the UI.
Everything looks good from my side.
<img width="1710" alt="Screenshot 2025-04-28 at 7 20 49 PM" src="https://github.com/user-attachments/assets/9bf19d4b-1b1c-4664-ad59-c4566f368e5a" />

agamv25 commented:
Hi Lachlan, I was able to run the program successfully on my windows machine
Everything seems to be working correctly as intended
![image](https://github.com/user-attachments/assets/dfef34e3-a4d3-4e5a-8328-b8c7bef024ed)

atharv02-git reviewed (approved):
Hello again @lachlan-robinson, just went through your work, and below are my findings:

- [ ] Firstly, I was able to pull and rebuild the container with your work without generating any new errors.
- [ ] The new `Progress-Burndown-Chart-component` has been imported properly inside the doubtfire-angular.module and doubtfire-angularjs.module and all the older instances has been removed/downgraded.
- [ ] Property binding style to use the Angular form, has been updated in all the necessary components which import `progress-dashboard` component.
- [ ] old coffee has been removed from `directives.coffee`
- [ ] `progress-dashboard.coffee` has been deleted
- [ ] In the old `progress-dashboard.tpl.html`, charts were rendered using the `<f-progress-burndown-chart> and <f-task-visualisation> components`. The component tags suggest that custom chart components were being used to display the progress burndown and task statuses.
- [ ] The new HTML uses Angular-specific components `<mat-form-field>, <mat-select>` with Material Design to improve the structure, functionality, and styling of the user interface. The charts are still being displayed via the same custom components `<f-progress-burndown-chart>, <f-task-visualisation>`, but the way the UI is organized has been enhanced.

I approve these changes, and great work with closing that loop, from migrating all the necessary services `visualisation, listener` which is no longer needed in new components but it is still being fetched by other components which adds an impact on how deep your research was in order to migrate the final component (progress-dashboard). Keep up the good work brother 🫡👌

aNebula reviewed (approved):
LGTM!
@lachlan-robinson Please open a upstream PR against [doubtfire-lms/doubtfire-web 9.x branch](https://github.com/doubtfire-lms/doubtfire-web/tree/9.x).

returnMarcco committed:
Migrate/group set selector (#981)

* chore: migrate group-set-selector (WIP)

- define and declare new component files

- import new component into app

- downgrade component

* chore: migrate group-set-selector

- convert AngularJS template syntax to Angular syntax
- finish implementing `group-set-selector.component.ts`
- remove redundant Coffeescript file

* chore: migrate group-set-selector

- finish defining Typescript logic
- update parent component to use Angular syntax for child component declaration

* chore: cleanup and documentation

- remove redundant Coffeescript file
- adjust frontend migration todo list

* chore: remove redundant file

- remove redundant Coffeescript file

* fix: attribute and event binding

- fix attribute and event binding syntax
- remove redundant template file

* refactor: two-way binding

- remove redundant directive attributes

- rename EventEmitter var name in child component

* Update README.md

* Update README.md

* refactor: replace Bootstrap components with Angular Material

- replace Bootstrap components with Angular Material components & directives

- refactor to use modern Angular structural directives

returnMarcco committed:
refactor: migrate/date service (#931)

* chore: date-service.coffee to TS

- add `date.service.ts` file
- add `date.service.spec.ts` unit test
- downgrade TS module such that it works within AngularJS

* chore: unlink date-service.coffee

- complete unlinking process of  `date-service.coffee` after migrating to Typescript
- manual linting/comment formatting

* chore: delete redundant coffeescript file

- delete `date-service.coffee` file after migration

* chore:  formatting

- format code with prettier

b0ink committed:
Merge branch '9.x' into migrate-progress-dashboard
